products/apigateway/api.yaml

```diff
@@ -0,0 +1,227 @@
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://apigateway.googleapis.com/v1beta/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: True
+    allowed:
+      - True
+      - False
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Api'
+    min_version: beta
+    create_url: projects/{{project}}/locations/global/apis?apiId={{api_id}}
+    self_link: projects/{{project}}/locations/global/apis/{{api_id}}
+    base_url: projects/{{project}}/locations/global/apis
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      A consumable API that can be used by multiple Gateways.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/api-gateway/docs/quickstart'
+      api: 'https://cloud.google.com/api-gateway/docs/reference/rest/v1beta/projects.locations.apis'
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: apiId
+        description: |
+          This must be unique within the project.
+        required: true
+        input: true
+        url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The resource name of the API. Format `projects/{{project}}/locations/global/apis/{{apiId}}`
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        description: |
+          A user-visible name for the API.
+      - !ruby/object:Api::Type::String
+        name: 'managedService'
+        input: true
+        description: |
+          Immutable. The name of a Google Managed Service ( https://cloud.google.com/service-infrastructure/docs/glossary#managed).
+          If not specified, a new Service will automatically be created in the same project as this API.
+      - !ruby/object:Api::Type::Time
+        name: 'createTime'
+        description: Creation timestamp in RFC3339 text format.
+        output: true
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'labels'
+        description: |
+          Resource labels to represent user-provided metadata.
+  - !ruby/object:Api::Resource
+    name: 'ApiConfig'
+    min_version: beta
+    create_url: projects/{{project}}/locations/global/apis/{{api}}/configs?apiConfigId={{api_config_id}}
+    self_link: projects/{{project}}/locations/global/apis/{{api}}/configs/{{api_config_id}}
+    base_url: projects/{{project}}/locations/global/apis/{{api}}/configs
+    read_query_params: '?view=FULL'
```

Comment:
Have you tried just setting this on `self_link`? If that works, then you can avoid having to add an extra MM attribute

Reply:
I haven't but I generated one now and it doesn't work. It replaced all the urls including Create and Delete methods

```
2020/10/21 19:23:23 [DEBUG] Google API Request Details:
---[ REQUEST ]---------------------------------------
DELETE /v1beta/projects/REDACTED/locations/global/apis/test/configs/tf-test-api-cfgwn1ovme2i3?alt=json&view=FULL HTTP/1.1
Host: apigateway.googleapis.com
User-Agent: Terraform/0.12.29 (+https://www.terraform.io) Terraform-Plugin-SDK/2.0.3 terraform-provider-google-beta/acc 
Content-Type: application/json
Accept-Encoding: gzip


-----------------------------------------------------
2020/10/21 19:23:24 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 400 Bad Request
Connection: close
Transfer-Encoding: chunked
Alt-Svc: h3-Q050=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-27=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-T050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Wed, 21 Oct 2020 18:23:24 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

229
{
  "error": {
    "code": 400,
    "message": "Invalid JSON payload received. Unknown name \"view\": Cannot bind query parameter. Field 'view' could not be found in request message.",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.BadRequest",
        "fieldViolations": [
          {
            "description": "Invalid JSON payload received. Unknown name \"view\": Cannot bind query parameter. Field 'view' could not be found in request message."
          }
        ]
      }
    ]
  }
}

0

```

Reply:
I'm going to go with the mm attribute.